docs/TOKENS.md

```diff
@@ -13,133 +13,141 @@ Spoilers. My love for Astro has grown deeper.
+These are the only two valid inputs. If one of those inputs is in the URL, we can assign an `agoraRole` variable to an enum predefined within the `agora-token` package. If the request does not include one of these options, we return a Bad Request using our helper function.
 
 ```ts
 import agoraToken from "agora-token";
 
-export async function GET({ params }: APIContext) {
-    //check for valid channel
-    if (!params.channel) {
-        return new Response("channel is required", { status: 400, headers })
+export async function POST({ request }: APIContext) {
+    const { channel, role, uid, expireTime } = await request.json()
+
+    if (!channel) {
+        return sendBadRequest("channel is required")
     }
-    //check for valid role
-    let role;
-    if (params.role === 'publisher') {
-        role = agoraToken.RtcRole.PUBLISHER;
-    } else if (params.role === 'subscriber') {
-        role = agoraToken.RtcRole.SUBSCRIBER
-    } else {
-        return new Response("role is incorrect", { status: 400, headers })
+    if (!uid) {
+        return sendBadRequest("uid is required")
     }
-    //check for valid uid
-    if (!params.uid || params.uid === '') {
-        return new Response("uid is required", { status: 400, headers })
+    if (!expireTime) {
+        return sendBadRequest("expireTime is required")
+    }
+
+    let agoraRole;
+    if (role === 'publisher') {
+        agoraRole = agoraToken.RtcRole.PUBLISHER;
+    } else if (role === 'subscriber') {
+        agoraRole = agoraToken.RtcRole.SUBSCRIBER
+    } else {
+        return sendBadRequest("role is incorrect")
     }
-  return new Response(
-    JSON.stringify({
-      rtcToken: "token",
-    })
-  )
+
+    return new Response(JSON.stringify({
+        token: "token"
+    }), { status: 200 })
 }
 ```
 
 ## Generate Token
-The last piece of information we need to be able to generate our token is the token expiration time and the privilege expiration time. These are in seconds, so I will set them to 10 minutes (600 seconds). You can set this to whatever time is appropriate for your use case.
+We will create a separate `handleGenerateToken` function to generate the token. This logic will allow us to create a token within the same repository without needing to use a `POST` request.
+
+To create the token, we use the `buildTokenWithUid` function from the `agora-token` package. We pass `expireTime` to the token and privilege expiration field.
+
+The privilege expiration time defines how long users will have their privileges within the channel. If the privilege time expires, the user can still listen to the call but can no longer publish. Once the token expires, they will lose access to the channel completely. In our case, we will set the token and privilege expiration to the same duration.
```

Comment:
Add a short explanation about the token expiration (similar to privilege) . Currently the user needs to extrapolate that piece of information when you should make it clear, before explaining what happens when each one expires.

> The privilege expiration time defines how long users will have their privileges within the channel, while token expiration defines how long users can remain in the channel.

Reply:
I thought the 3rd sentence did that, but I added in your suggestion too, since I don't think it hurts